File: Java (client)/ClientServerWithJava/Service/src/main/java/ro/mpp2024/IService.java
package ro.mpp2024;

import java.util.Collection;
import java.util.Optional;

public interface IService <ID>{
    //boolean addUser(String name, String password);

    //User connect(String name, String password);

    Collection<String> getListMatches(String search);

    Optional<User> login(String username, String password, Observer client);

    void logout(String username);

    MatchDTO getMatch(int i);

    boolean addTicket(BuyTicketsDTO ticket);
}

File: C# (server)/ClientServerWithJava/Service/IService.cs
using System;
using System.Collections.Generic;
using Model;

namespace Service
{
    public interface IService
    {
        ICollection<string> GetListMatches(string search);

        User Login(string username, string password, Observer client);

        void Logout(string username);

        MatchDTO GetMatch(int i);

        bool AddTicket(BuyTicketsDTO ticket);
    }
}